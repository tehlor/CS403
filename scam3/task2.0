(define (abs n) (if (< n 0) (- n) n))
(replace abs (list '< < '- -))
(inspect (get 'code abs))
