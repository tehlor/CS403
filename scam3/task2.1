(define (square x) (* x x))
(replace square (list '* +))
(inspect (square 5))
