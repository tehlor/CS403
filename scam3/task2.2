(include "pretty.lib")
(define (square x) (* x x))
(replace square (list 'x 'y))
(pretty square)
(inspect (square 48))
