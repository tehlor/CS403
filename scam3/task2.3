(include "pretty.lib")
(define (almost-square x) ((lambda (y) (* y y)) (+ x 1)))
(inspect (almost-square 8))
(replace almost-square (list '* * '+ +))
(pretty almost-square)
(inspect (almost-square 8))
