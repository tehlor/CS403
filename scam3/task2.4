(include "pretty.lib")
(define (almost-square x) ((lambda (y) (* y y)) (+ x 1)))
(inspect (almost-square 15))
(replace almost-square (list '* + '+ *))
(pretty almost-square)
(inspect (almost-square 15))
