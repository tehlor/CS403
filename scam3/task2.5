(include "pretty.lib")
(define (almost-square x)
    ((lambda (y)
        (inspect (quote (lambda (z) (+ (* x y) z))))
        (* y y)) (+ x 1)))
(inspect (almost-square 24))
(replace almost-square (list 'quote quote '* * '+ +))
(pretty almost-square)
(inspect (almost-square 24))
