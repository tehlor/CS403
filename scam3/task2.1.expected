(square 5) is 10
